src/ripple/basics/base_uint.h

```diff
@@ -126,25 +126,45 @@ class base_uint
         *this = beast::zero;
     }
 
-    explicit base_uint (Blob const& vch)
+    explicit base_uint (std::uint64_t b)
     {
-        assert (vch.size () == size ());
-
-        if (vch.size () == size ())
-            memcpy (pn.data (), vch.data (), size ());
-        else
-            *this = beast::zero;
+        *this = b;
     }
 
-    explicit base_uint (std::uint64_t b)
+    template <class Container,
+              class = std::void_t
+              <
+                  decltype(std::declval<Container const>().size()),
+                  decltype(std::declval<Container const>().data()),
+                  typename Container::value_type
+              >,
+              class = std::enable_if_t
+              <
+                  std::is_trivially_copyable<typename Container::value_type>::value
+              >>
+    explicit base_uint(Container const& c)
     {
-        *this = b;
+        assert(c.size()*sizeof(typename Container::value_type) == size());
+        std::memcpy(pn.data(), c.data(), size());
     }
 
-    template <class OtherTag>
-    void copyFrom (base_uint<Bits, OtherTag> const& other)
+    template <class Container,
+              class = std::void_t
+              <
+                  decltype(std::declval<Container const>().size()),
+                  decltype(std::declval<Container const>().data()),
+                  typename Container::value_type
+              >>
+     std::enable_if_t
+     <
+         std::is_trivially_copyable<typename Container::value_type>::value,
+         base_uint&
+     >
+    operator=(Container const& c)
     {
-        memcpy (pn.data (), other.data(), bytes);
+        assert(c.size()*sizeof(typename Container::value_type) == size());
+        std::memcpy(pn.data(), c.data(), size());
+        return *this;
     }
 
     /* Construct from a raw pointer.
```

Comment:
Bummer.  I really wish we could remove `fromVoid()`.  I looked at it a bit.  There are 4 uses of `fromVoid()` left in the code base.  Two of them are in unit tests and could easily be fixed.  Unfortunately the other two are harder to fix.  They are:

-  https://github.com/ripple/rippled/blob/develop/src/ripple/nodestore/backend/MemoryFactory.cpp#L127
- https://github.com/ripple/rippled/blob/develop/src/ripple/nodestore/impl/DecodedBlob.cpp#L85

For these two, the available information really is a `void*`.  Those two will have to be fixed before `fromVoid()` can be removed.  I'm uncertain how complicated those fixes will be, but they're not trivial local changes.

Reply:
I had overlooked the `fromVoid()` in tests.  Fixing those, thanks.